app/lib/features/pins/pages/pin_page.dart

```diff
@@ -119,117 +115,69 @@ class PinPage extends ConsumerWidget {
     }
 
     return PopupMenuButton(
+      icon: const Icon(Atlas.dots_vertical_thin),
       itemBuilder: (ctx) => actions,
     );
   }
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    // ignore: unused_local_variable
     final pin = ref.watch(pinProvider(pinId));
+
     return Scaffold(
       body: CustomScrollView(
         slivers: <Widget>[
-          PageHeaderWidget(
-            title: pin.hasValue ? pin.value!.title() : 'Loading pin',
-            sectionDecoration: const BoxDecoration(
-              gradient: primaryGradient,
-            ),
-            actions: [
-              pin.maybeWhen(
-                data: (pin) => buildActions(context, ref, pin),
-                orElse: () => const SizedBox.shrink(),
-              ),
-            ],
-          ),
           pin.when(
-            data: (pin) {
-              final isLink = pin.isLink();
-              final spaceId = pin.roomIdStr();
-              final List<Widget> content = [];
-              if (isLink) {
-                content.add(
-                  OutlinedButton.icon(
-                    icon: const Icon(Atlas.link_chain_thin),
-                    label: Text(pin.url() ?? ''),
-                    onPressed: () async {
-                      final target = pin.url()!;
-                      await openLink(target, context);
-                    },
+            data: (acterPin) {
+              final pinEdit = ref.watch(pinEditProvider(acterPin));
+              final pinEditNotifer =
+                  ref.watch(pinEditProvider(acterPin).notifier);
+              return SliverAppBar(
+                automaticallyImplyLeading: false,
+                centerTitle: false,
+                leadingWidth: 40,
+                toolbarHeight: 100,
+                flexibleSpace: Container(
+                  decoration: const BoxDecoration(gradient: primaryGradient),
+                ),
+                leading: IconButton(
+                  onPressed: () => context.canPop()
+                      ? context.pop()
+                      : context.goNamed(Routes.pins.name),
+                  icon: const Icon(
+                    Icons.chevron_left,
+                    size: 42,
                   ),
```

Comment:
```suggestion
                automaticallyImplyLeading: true,
                centerTitle: false,
                leadingWidth: 40,
                toolbarHeight: 100,
                flexibleSpace: Container(
                  decoration: const BoxDecoration(gradient: primaryGradient),
                ),
```

Don't provide a custom `leading` if it isn't necessary. And it isn't necessary. Always prefer `automaticallyImplyLeading` over adding custom local routing and designs as they are forgotten easily when updates need to happen and then lead to weird custom behaviors.